include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -187,6 +187,37 @@ def XTenNN_LoadExternalConstOp: XTenNN_Op<"load_external_const", [
 
 def TosaExtension : NativeOpTrait<"TosaExtension">;
 def ElementwiseUnary : NativeOpTrait<"ElementwiseUnary">;
+class ArrayAttr<int n> : ConfinedAttr<ArrayAttr, [ArrayCount<n>]>;
+class I64DenseArrayAttr<int n> : ConfinedAttr<DenseI64ArrayAttr, [DenseArrayCount<n>]>;
+
+def XtenNN_GroupConv2dOp: XTenNN_Op<"group_conv2d", [Pure, TosaExtension]> {
+  let summary = "Grouped convolution with multiple channel outputs per layer";
+  let description = [{
+    Calculates separates convolutions with multiple channel outputs per layer. The number
+    of groups need to be higher than one.
+
+    This operations expects the same TORCH/ONNX's Conv2d datalayout specification for input and
+    weight [NxCxHxW].
+
+    Pad attribute is expected to be two arrays of i64 values, eg: [[0,1], [2,2]]. First and second
+    arrays denote the initial and end padding for H and W dimensions respectively.
+  }];
+  let arguments = (ins
+    4DTensorOf<[BF16, F32]>:$input,
+    4DTensorOf<[BF16, F32]>:$weights,
+    1DTensorOf<[BF16, F32]>:$bias,
+    ArrayAttr<2>:$pad,
+    I64DenseArrayAttr<2>:$stride,
+    I64DenseArrayAttr<2>:$dilation,
+    I64Attr:$group
+  );
+  let results = (outs
+    AnyTensor:$output
```

Comment:
Should be same type of Tensor as `input`?

Reply:
Right! Done